### PR TITLE
【KernelGen】Add cudnn_convolution operator

### PR DESCRIPTION
**Pull request by Schopenhauer-loves-Hegel**
### PR Category
Operator

### Type of Change
New Feature

### Description
Add `cudnn_convolution` operator implementation with Triton kernel.

- Implementation mode: `N/A`
- Accuracy test: 16/16 passed

### Issue
N/A

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### Performance
_Benchmark data not available._

---
_Generated by auto_gen tool with Claude Code_

### CODE REVIEW

**Comment by Schopenhauer-loves-Hegel** [comment]
Closing in favor of grouped PRs to reduce review overhead.

File: tests/test_cudnn_convolution.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+]
+
+
+@pytest.mark.cudnn_convolution
+@pytest.mark.parametrize("shape, kernel, groups", SHAPE_CUDNN_CONV2D)
+@pytest.mark.parametrize("stride", [1, 2])
+@pytest.mark.parametrize("padding", [0, 1])
+@pytest.mark.parametrize("dtype", [torch.float16, torch.float32])
+@pytest.mark.parametrize("dilation", [1, 2])
+def test_accuracy_cudnn_convolution_2d(
```
Comment (tengqm) [inline tests/test_cudnn_convolution.py]:
```suggestion
def test_cudnn_convolution_2d(
```

File: tests/test_cudnn_convolution.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    ((64, 64, 64), (128, 64, 7)),
+]
+
+
+@pytest.mark.cudnn_convolution
+@pytest.mark.parametrize("shape, kernel", SHAPE_CUDNN_CONV1D)
+@pytest.mark.parametrize("stride", [1, 2])
+@pytest.mark.parametrize("padding", [0, 1])
+@pytest.mark.parametrize("dtype", [torch.float16, torch.float32])
+def test_accuracy_cudnn_convolution_1d(
```
Comment (tengqm) [inline tests/test_cudnn_convolution.py]:
```suggestion
def test_cudnn_convolution_1d(
```

File: tests/test_cudnn_convolution.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+]
+
+
+@pytest.mark.cudnn_convolution
+@pytest.mark.parametrize("shape, kernel, groups", SHAPE_CUDNN_CONV3D)
+@pytest.mark.parametrize("stride", [1, 2])
+@pytest.mark.parametrize("padding", [0, 1])
+@pytest.mark.parametrize("dtype", [torch.float16, torch.float32])
+@pytest.mark.parametrize("dilation", [1, 2])
+def test_accuracy_cudnn_convolution_3d(
```
Comment (tengqm) [inline tests/test_cudnn_convolution.py]:
```suggestion
def test_cudnn_convolution_3d(
```

**Review by tengqm — DISMISSED** [review]

File: benchmark/test_cudnn_convolution_perf.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import pytest
+import torch
+
+import flag_gems
+from benchmark.attri_util import FLOAT_DTYPES
+from benchmark.performance_utils import GenericBenchmark
```
Comment (tengqm) [inline benchmark/test_cudnn_convolution_perf.py]: Rebase to master

Comment (factnn) [reply benchmark/test_cudnn_convolution_perf.py]: Done

File: conf/operators.yaml [hunk 5]
```diff
@@ -6159,3 +5406,15 @@ ops:
...
+      A wrapper for cuDNN convolution backend.
+    for:
+      - cudnn_convolution
+    labels:
+      - aten
+      - KernelGen
+    kind:
+      - NeuralNetwork
+    stages:
+      - beta: '5.1'
```
Comment (tengqm) [inline conf/operators.yaml]: Where is the performance benchmark?

**Review by w1120029931-bit — CHANGES_REQUESTED** [review]
please change file test_cudnn_convolution_perf.py to file test_cudnn_convolution.py

**Review by w1120029931-bit — APPROVED** [review]
